### ISSUE TITLE
Hybrid search with remote embedding doesn't return relevant keyword results.

### ISSUE THREAD

**Issue by capybarahero** [typesense/typesense#2060]
## Description

We cannot run a hybrid search with remote embedding and "prefix: true". The requirement for "prefix: false" impacts both keyword and semantic search, ignoring relevant documents that would be found via keyword.

## Steps to reproduce

Option (1): Create a hybrid search using remote embedding and setting "prefix: true". 

This will lead to "_Prefix search is not supported for remote embedders. Please set `prefix=false` as an additional search parameter to disable prefix searching._" as expected.

Option (2): Create a hybrid search using remote embedding and setting "prefix: false" as the example below:

```
{
    "collection": "users",    
    "prefix": false,    
    "query_by": "full_name,vector",
    "vector_query": "vector:([], k:50)",
    "q": "jo",
    }
```
This will correctly return results based on the embedding. However, a person with full_name "John" will not be returned.

## Expected Behavior

Running a hybrid search with remote embedding should not impact valid keyword search results.

## Actual Behavior

Relevant keyword results are ignored.

## Metadata

**Typesense Version**: v27.1

**OS**: n/a

**Comment by lovlyx** [comment]
Any updates on this?

**Comment by Jackman3005** [comment]
Yeah, I was bummed to find this out as well. I'm assuming the reason for this is technical, although I'm not quite sure why remote vs local (assuming) makes a difference for ability to use prefix.

I don't think embeddings have the ability to do prefixing regardless, right? Or at least not explicitly, instead it's just handled implicitly or not at all I would think. How does this change how the other half of hybrid search works though? Or maybe I just don't understand how prefix works...

**Comment by MariusWilsch** [comment]
Any updates on this?
